app/functoria_app.ml

```diff
@@ -622,36 +624,64 @@ module Make (P: S) = struct
     List.fold_left (fun r p -> r >>= fun () -> Bos.OS.Path.delete p) (R.ok ()) files >>= fun () ->
-    with_current config_dir
+    with_current build_dir
       (fun () ->
-         let pkgs = match P.packages with
-           | []   -> Bos.Cmd.empty
-           | pkgs ->
-             let pkgs =
-               List.fold_left (fun acc pkg ->
-                   String.Set.union pkg.ocamlfind acc
-                 ) String.Set.empty pkgs
-               |> String.Set.elements
-             in
-             Bos.Cmd.(v "-pkgs" % String.concat ~sep:"," pkgs)
-         in
-         let ignore_dirs = match P.ignore_dirs with
-           | []    -> Bos.Cmd.empty
-           | dirs  -> Bos.Cmd.(v "-Xs" % String.concat ~sep:"," dirs)
-         in
-         let cmd =
-           Bos.Cmd.(v "ocamlbuild" % "-use-ocamlfind" % "-classic-display" %
-                    "-tags" % "bin_annot" % "-quiet" %%
-                    ignore_dirs %% pkgs % file)
+          Log.info (fun m -> m "%s" file_ml);
+          let target_dir = Fpath.(v "build-config")
+          (* Import files in the build target *)
+          and target_path = Fpath.(v "build-config" / file_ml)
+          and target_path_dune = Fpath.(v "build-config" / "config.dune")
+          in
+          Bos.OS.Dir.create target_dir >>= fun _ ->
+          Bos.OS.Path.symlink ~force:true ~target:(Fpath.(v ".." // config_dir_to_build_dir / file_ml)) target_path >>= fun () ->
+          Bos.OS.Path.exists Fpath.(config_dir_to_build_dir / "config.dune") >>= fun res ->
+          (match res with
+          | true ->  (Bos.OS.Path.symlink ~force:true ~target:(Fpath.(v ".." // config_dir_to_build_dir / "config.dune")) target_path_dune >>= fun () -> Ok true)
+          | false -> Ok false)
+          (* Generate dune configuration file *)
+          >>= fun has_dune_inc ->
+          let dune_file = Fpath.(v "build-config" / "dune")
+          and dune_project_file = Fpath.(v "dune-project") in
+          let pkgs = match P.packages with
+            | []   -> ""
+            | pkgs ->
+              let pkgs =
+                List.fold_left (fun acc pkg ->
+                    String.Set.union pkg.ocamlfind acc
+                  ) String.Set.empty pkgs
+                |> String.Set.elements
+              in
+              let pkgs = if has_dune_inc then "config_custom"::pkgs else pkgs
+              in
+              String.concat ~sep:" " pkgs
+          in
+          let dune_content_default = "(library (name config) (modules config) (libraries "^pkgs^"))" in
+          let dune_content = if has_dune_inc then ("(include config.dune)"^dune_content_default) else dune_content_default in
+          let write_dune_file = Bos.OS.File.delete dune_file >>= fun () -> Bos.OS.File.write dune_file dune_content
+          and write_dune_project_file = Bos.OS.File.exists dune_project_file 
+          >>= function | false -> Bos.OS.File.write dune_project_file "(lang dune 1.7)" 
```

Comment:
1.0 would be a more conservative choice if you don't need fancy features.

Reply:
Yes, I updated that !